food-truck-api/src/main/java/food/truck/api/endpoint/TruckEndpoint.java

```diff
@@ -58,69 +58,63 @@ public String postTruckReview(@AuthenticationPrincipal User user, @PathVariable
 
     @Value
     private static class CreateTruckParams {
-        @NonNull
-        long userId;
-        @NonNull
-        String token;
         @Nullable
         String truckName;
     }
 
     @PostMapping("/truck/create")
-    public Truck createTruck(@RequestBody CreateTruckParams data) {
-        return truckService.createTruck(data.userId, data.truckName);
-    }
-
-    @Value
-    private static class DeleteTruckParams {
-        @NonNull
-        long userId;
-        @NonNull
-        String token;
-        @NonNull
-        long truckId;
+    public Truck createTruck(@AuthenticationPrincipal User u, @RequestBody CreateTruckParams data) {
+        return truckService.createTruck(u.getId(), data.truckName);
     }
 
-    @DeleteMapping("/truck/delete")
-    public void deleteTruck(@RequestBody DeleteTruckParams data) {
-        truckService.deleteTruck(data.truckId);
+    @DeleteMapping("/truck/delete/{truckId}")
+    public void deleteTruck(@AuthenticationPrincipal User u, @PathVariable long truckId) {
+        var t = truckService.findTruckById(truckId);
+        t.ifPresent(truck -> {
+            if (truck.getUserId().equals(u.getId())) {
```

Comment:
Didn't think of that edge case, but letting some random stranger take stabs in the dark and eventually delete one of our food trucks would have been bad. Good catch!